benchmarks/definitions/nashpobench.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from pathlib import Path
+import numpy as np
+from random import shuffle
+
+from sklearn.ensemble import RandomForestRegressor
+
+from syne_tune.search_space import choice, randint, uniform, loguniform
+
+from blackbox_repository.conversion_scripts.scripts.fcnet_import import \
+    METRIC_ELAPSED_TIME, METRIC_VALID_LOSS, RESOURCE_ATTR, BLACKBOX_NAME
+
+
+class SubsamplingSurrogate(object):
+
+    def __init__(self, ratio=0.01):
+        self.ratio = ratio
+
+    def fit(self, X, y):
+        n = int(X.shape[0] * self.ratio)
+        idx = [i for i in range(X.shape[0])]
+
+        shuffle(idx)
+        idx = idx[:n]
+        X_train = X[idx]
+        y_train = np.array(y)[idx]
+
+        self.model = RandomForestRegressor()
+        self.model.fit(X_train, y_train)
+        return self
+
+    def predict(self, X_test):
+        return self.model.predict(X_test)
+
+
+_config_space = {
+    "hp_activation_fn_1": choice(["tanh", "relu"]),
+    "hp_activation_fn_2": choice(["tanh", "relu"]),
+    "hp_batch_size": randint(8, 64),
+    "hp_dropout_1": uniform(0.0, 0.6),
+    "hp_dropout_2": uniform(0.0, 0.6),
+    "hp_init_lr": loguniform(0.0005, 0.1),
+    'hp_lr_schedule': choice(["cosine", "const"]),
+    'hp_n_units_1': randint(16, 512),
+    'hp_n_units_2': randint(16, 512),
+}
+
+
+def nashpobench_default_params(params=None):
+    dont_sleep = str(
+        params is not None and params.get('backend') == 'simulated')
+    return {
+        'max_resource_level': 100,
+        'grace_period': 1,
+        'reduction_factor': 3,
+        'instance_type': 'ml.m5.large',
+        'num_workers': 4,
+        'framework': 'PyTorch',
+        'framework_version': '1.6',
+        'dataset_name': 'protein_structure',
+        'dont_sleep': dont_sleep,
+        'cost_model_type': 'linear',
+    }
+
+
+def nashpobench_benchmark(params):
+    """
+    The underlying tabulated blackbox does not have an `elapsed_time_attr`,
+    but only a `time_this_resource_attr`.
+
+    """
+    config_space = dict(
+        _config_space,
+        epochs=params['max_resource_level'],
+        dataset_name=params['dataset_name'],
+        dont_sleep=params['dont_sleep'],
+        blackbox_repo_s3_root=params.get('blackbox_repo_s3_root'))
+    return {
+        'script': Path(__file__).parent.parent.parent / "examples" / "training_scripts" / "nashpobench" / "nashpobench.py",
```

Comment:
As discussed with David, we may not want to have this for every tabulated blackbox.

The plus is we can then use the blackbox also with other back-ends, not just simulated. But it is not sure whether we want that.

Reply:
I agree, I left it in there for future experiments regarding the simulator backend paper

Reply:
That makes sense, because that we need to do for a paper.

Reply:
If something is only needed for a paper, can we just have it in a branch? (the cost of maintaining code is much higher than the writing cost)

Reply:
fair point, I will drop the file and add it to a separated branch

Reply:
If you remove the script in examples/training_scripts, you can set 'script' to None here, that should work. But please test it, I am not sure.

Reply:
Howeever, we should keep the nasbench201 script in examples/training_scripts, if only as a reminder that we'd like to have a general wrapper script.